scripts/testing.sh

```diff
@@ -1,12 +1,41 @@
 #!/bin/bash
 
-set -e 
+set -x
+export INTERNAL_REGISTRY=${INTERNAL_REGISTRY:-"default-route-openshift-image-registry.apps.testocp47.psiodo.net"}
+export IMAGE_NAMESPACE="odoinitimage$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 4 | head -n 1)"
+export IMAGE_NAME=initimage
+export IMAGE=$INTERNAL_REGISTRY/$IMAGE_NAMESPACE/$IMAGE_NAME
 
-# Get master Odo
-git clone https://github.com/openshift/odo $GOPATH/src/github.com/openshift/odo
+#login
+#login as developer on openshift cluster
+set +x
+oc login -u developer -p ${OCP4X_DEVELOPER_PASSWORD} --insecure-skip-tls-verify ${OCP4X_API_URL}
+#login as developer with podman
+podman login --tls-verify=false -u developer -p $(oc whoami -t) $INTERNAL_REGISTRY
+set -x
 
-# Retrieve the version / what's currently being used as SupervisorD
-IMAGE=`cat $GOPATH/src/github.com/openshift/odo/pkg/devfile/adapters/common/utils.go | grep "defaultBootstrapperImage = " | cut -d \" -f2 | sed '/^\s*$/d'`
+#create project for the image
+oc new-project $IMAGE_NAMESPACE
 
-# Build the container
-docker build -t $IMAGE .
+#build odo-init-image
+. ./scripts/build-push-image.sh
+
+echo "Exporting ODO_BOOTSTRAPPER_IMAGE"
+export ODO_BOOTSTRAPPER_IMAGE="$IMAGE"
+
+# Create temporary directory for cloning odo repo
+RANDOM_DIR=$(mktemp -d)
+pushd $RANDOM_DIR
+
+#clone odo repo for testing
+git clone https://github.com/openshift/odo $RANDOM_DIR
+
+#----
+
+# execute tests
+sh -c ". ./scripts/setup_script_e2e.sh && . ./scripts/run_script_e2e.sh"
+
+#cleanup
+popd
+rm -rf $RANDOM_DIR
+oc delete project $IMAGE_NAMESPACE
```

Comment:
If scripts error out in between this namespace will also left out in the cluster. Don't you think we should add same label `odo` here as well.

Reply:
yes, I will add labels to this namespace too.
Thank you for pointing this out.